npms/skale-observer/observer.mjs

```diff
@@ -0,0 +1,1056 @@
+            }
+        };
+        gClient.send( jo );
+        return true;
+    } catch ( err ) {
+        if( log.verboseGet() >= log.verboseReversed().error ) {
+            log.write( cc.error( "Failed to start parallel periodic SNB refresh, error is: " ) +
+                cc.warning( owaspUtils.extractErrorMessage( err ) ) +
+                cc.error( ", stack is: " ) + "\n" + cc.stack( err.stack ) + "\n" );
+        }
+    }
+    return false;
+}
+
+export async function periodicCachingStart( strChainNameConnectedTo, addressFrom, opts ) {
+    gFlagHaveParallelResult = false;
+    const bParallelMode =
+        ( opts && "bParallelMode" in opts &&
+        typeof opts.bParallelMode != "undefined" &&
+        opts.bParallelMode )
+            ? true : false;
+    let wasStarted = false;
+    if( bParallelMode ) {
+        wasStarted =
+            parallelPeriodicCachingStart( strChainNameConnectedTo, addressFrom, opts );
+    }
+    if( wasStarted )
+        return;
+    inThreadPeriodicCachingStart( strChainNameConnectedTo, addressFrom, opts );
+}
+
+export async function periodicCachingStop() {
+    if( gWorker && gClient ) {
+        try {
+            const jo = {
+                "method": "periodicCachingStop",
+                "message": { }
+            };
+            gClient.send( jo );
+        } catch ( err ) {
+            if( log.verboseGet() >= log.verboseReversed().error ) {
+                log.write( cc.error( "Failed to stop parallel periodic SNB refresh, error is: " ) +
+                    cc.warning( owaspUtils.extractErrorMessage( err ) ) +
+                    cc.error( ", stack is: " ) + "\n" + cc.stack( err.stack ) + "\n" );
+            }
+        }
+    }
+    if( gIntervalPeriodicCaching ) {
+        try {
+            clearInterval( gIntervalPeriodicCaching );
+            gIntervalPeriodicCaching = null;
+        } catch ( err ) {
+            if( log.verboseGet() >= log.verboseReversed().error ) {
+                log.write( cc.error( "Failed to stop in-thread periodic SNB refresh, error is: " ) +
+                    cc.warning( owaspUtils.extractErrorMessage( err ) ) +
+                    cc.error( ", stack is: " ) + "\n" + cc.stack( err.stack ) + "\n" );
+            }
+            gIntervalPeriodicCaching = null; // clear it anyway
+        }
+    }
+    gFlagHaveParallelResult = false;
+}
+
+export function pickRandomSChainNodeIndex( joSChain ) {
+    let min = 0, max = joSChain.data.computed.nodes.length - 1;
+    min = Math.ceil( min );
+    max = Math.floor( max );
+    const idxNode = Math.floor( Math.random() * ( max - min + 1 ) ) + min;
+    return idxNode;
+}
+export function pickRandomSChainNode( joSChain ) {
+    const idxNode = pickRandomSChainNodeIndex( joSChain );
+    return joSChain.data.computed.nodes[idxNode];
+}
+
+export function pickRandomSChainUrl( joSChain ) {
+    const joNode = pickRandomSChainNode( joSChain );
+    return "" + joNode.http_endpoint_ip;
```

Comment:
Change to camelCase

Reply:
Fixed, switched to `array[ "access_notation" ]` for comparibility with old similar JSON RPC answer format generated by `skaled`.